File: packages/ts-sdk/test/adapters/registry.test.ts
import { describe, expect, it } from 'vitest';
import { RegistryAdapter } from '../../src/treedx/index.js';
import { MockTransport } from './mock.js';

describe('RegistryAdapter', () => {
  it('constructs registry endpoints', async () => {
    const transport = new MockTransport();
    const adapter = new RegistryAdapter({ transport });
    await adapter.localNode();
    await adapter.nodes();
    await adapter.getPlacement('repo');
    await adapter.setPlacement('repo', {});
    expect(transport.requests.map((r) => `${r.method} ${r.path}`)).toEqual([
      'GET /api/v1/node',
      'GET /api/v1/registry/nodes',
      'GET /api/v1/registry/repos/repo/placement',
      'POST /api/v1/registry/repos/repo/placement'
    ]);
  });
});

File: packages/elixir-sdk/test/adapters/registry_test.exs
defmodule TreeDxSdk.RegistryAdapterTest do
  use ExUnit.Case, async: true

  test "all constructs expected request" do
    {:ok, pid} = TreeDxSdk.Test.MockTransport.start_link()
    client = TreeDxSdk.Test.MockTransport.client(pid)
    TreeDxSdk.Registry.local_node(client)
    TreeDxSdk.Registry.nodes(client)
    TreeDxSdk.Registry.get_placement(client, "repo/a")
    TreeDxSdk.Registry.set_placement(client, "repo/a", %{})

    assert Enum.any?(
             TreeDxSdk.Test.MockTransport.requests(pid),
             &(&1.method == :post and &1.path == "/api/v1/registry/repos/repo%2Fa/placement")
           )
  end
end
